src/Analyser/NodeScopeResolver.php

```diff
@@ -1148,6 +1148,7 @@ private function processStmtNode(
 			$branchScopeResult = $this->processStmtNodes($stmt, $stmt->stmts, $scope, $nodeCallback);
 			$branchScope = $branchScopeResult->getScope();
 			$finalScope = $branchScopeResult->isAlwaysTerminating() ? null : $branchScope;
+			$finalScope = $branchScopeResult->getScope()->mergeWith($finalScope);
```

Comment:
To me it looks like you're merging `$branchScope` with `$branchScope` or `null`. Are you sure?

Reply:
No, I'm merging `$branchScopeResult`'s `$finalScope` with `$finalScope`. This is the only way I could think of fixing this issue, is it proper?

Reply:
You might be right but I might be blind, but what I see is:

```php
$branchScope = $branchScopeResult->getScope();
$finalScope = $branchScopeResult->isAlwaysTerminating() ? null : $branchScope;
$finalScope = $branchScopeResult->getScope()->mergeWith($finalScope);
```

Which can be rewritten as:

```php
$branchScope = $branchScopeResult->getScope();
$finalScope = $branchScopeResult->isAlwaysTerminating() ? null : $branchScope;
$finalScope = $branchScope->mergeWith($finalScope); // $finalScope from previous line is null or $branchScope
// therefore you're doing either $branchScope->mergeWith($branchScope) or $branchScope->mergeWith(null)
```

Can you oppose what I see? :)

Reply:
Yeah, you're right. Pushed latest changes, in this case `$finalScope` is `null`, so in order to fix this assigning `$branchScope`'s scope to `$finalScope`.